// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTPeerReply";
option optimize_for = LITE_RUNTIME;

import public "BailmentReply.proto";
import public "ConnectionInfoReply.proto";
import public "OutBailmentReply.proto";
import public "NoticeAcknowledgement.proto";
import public "PeerEnums.proto";
import public "Signature.proto";

message PeerReply {
    optional uint32 version = 1;
    optional string id = 2;
    optional string initiator = 3;
    optional string recipient = 4;
    optional PeerRequestType type = 5;
    optional string cookie = 6;
    optional Signature signature = 7;
    optional string server = 8;
    oneof reply {
        BailmentReply bailment = 9;
        OutBailmentReply outbailment = 10;
        NoticeAcknowledgement notice = 11;
        ConnectionInfoReply connectioninfo = 12;
    }
}
